syntax = "proto3";

service HelloWorld{
  rpc SayHello(HelloRequest) returns (HelloReply) {}
}

message HelloRequest{
  string name = 1;
}

message HelloReply {
  string message = 1;
}

service Goodbye{
  rpc SayGoodbye(GoodbyeRequest) returns (GoodbyeReply) {}
}

message GoodbyeRequest {
  string name = 1;
}

message GoodbyeReply {
  string message = 1;
}

service Insult{
  rpc InsultPerson(InsultRequest) returns (InsultReply){}
}

message InsultRequest {
  string name = 1;
}

message InsultReply{
  string insult = 1;
}

service DatabaseCall {
  rpc DBCall(DatabaseRequest) returns (DatabaseReply){}
  rpc InfoFromID(InfoRequest) returns (InfoReply) {}
}

message DatabaseRequest {
  string firstname = 1;
}

message DatabaseReply {
  string lastname = 1;
}

message InfoRequest {
  int32 id = 1;
}

message InfoReply {
  string name = 1;
}